dace/frontend/python/newast.py

```diff
@@ -3973,11 +3974,19 @@ def parse_target(t: Union[ast.Name, ast.Subscript]):
         self._add_state('callback_%d' % node.lineno)
         self.last_state.set_default_lineinfo(self.current_lineinfo)
 
-        call_args = ', '.join(str(s) for s in allargs)
-        tasklet = self.last_state.add_tasklet(f'callback_{node.lineno}', {f'__in_{name}'
-                                                                          for name in args} | {'__istate'},
-                                              {f'__out_{name}'
-                                               for name in outargs} | {'__ostate'}, f'{funcname}({call_args})')
+        if callback_type.is_scalar_function() and len(callback_type.return_types) > 0:
```

Comment:
We should also consider cases where a callback has arguments and a return value. Perhaps in another PR.

Reply:
@alexnick83 testing tagging

AFAIU, `is_scalar_function` doesn't set limitation on the number of input arguments, just 0 or 1 scalar return values